applovin_max/ios/Classes/AppLovinMAX.m

```diff
@@ -393,6 +397,26 @@ - (void)setLocationCollectionEnabled:(BOOL)enabled
     }
 }
 
+- (void)setExtraParameter:(NSString *)key value:(NSString *)value
+{
+    if ( ![key al_isValidString] )
+    {
+        [self log: @"[%@] Failed to set extra parameter for nil or empty key: %@", TAG, key];
+        return;
+    }
+
+    if ( self.sdk )
+    {
+        ALSdkSettings *settings = self.sdk.settings;
+        [settings setExtraParameterForKey: key value: ( value != (id) [NSNull null] ) ? value : nil ];
```

Comment:
I don't think the cast to a `(id)` is necessary?
```suggestion
        [settings setExtraParameterForKey: key value: ( value != [NSNull null] ) ? value : nil];
```

Reply:
Without `(id)`, this warning is generated.  So it should be cast either `(id)` or `(NSString *)`.

```
Comparison of distinct pointer types ('NSString *' and 'NSNull * _Nonnull')
```

Reply:
Makes sense.